apps/hexagon_benchmarks/Makefile

```diff
@@ -0,0 +1,151 @@
+LDFLAGS-arm-32-profile-android ?= -llog -fPIE -pie
+
+BIN ?= bin
+
+FILTERS ?= conv3x3a16 dilate3x3 median3x3 gaussian5x5 sobel conv3x3a32
+
+OBJS = $(patsubst %,$(BIN)/\%/%_hvx64.o, $(FILTERS))
+OBJS += $(patsubst %,$(BIN)/\%/%_hvx128.o, $(FILTERS))
+OBJS += $(patsubst %,$(BIN)/\%/%_cpu.o, $(FILTERS))
+
+LINK_OBJS = $(patsubst %,$(BIN)/$$*/%_hvx64.o, $(FILTERS))
+LINK_OBJS += $(patsubst %,$(BIN)/$$*/%_hvx128.o, $(FILTERS))
+LINK_OBJS += $(patsubst %,$(BIN)/$$*/%_cpu.o, $(FILTERS))
+
+UPPERCASE_FILTERS = $(shell echo $(FILTERS) | tr '[:lower:]' '[:upper:]')
+DASH_D_DEFINES = $(patsubst %, -D%=1, $(UPPERCASE_FILTERS))
+
+
+$(BIN)/conv3x3: conv3x3.cpp $(GENERATOR_DEPS)
+	@-mkdir -p $(BIN)
+	$(CXX) $(CXXFLAGS) -O3 -g -fno-rtti $(filter-out %.h,$^) -o $@ $(LDFLAGS)
+
+$(BIN)/%/conv3x3a16_cpu.o: $(BIN)/conv3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f conv3x3a16_cpu target=$* accumulator_type=int16
+
+$(BIN)/%/conv3x3a16_hvx64.o: $(BIN)/conv3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f conv3x3a16_hvx64 target=$*-hvx_64 accumulator_type=int16
+
+$(BIN)/%/conv3x3a16_hvx128.o: $(BIN)/conv3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f conv3x3a16_hvx128 target=$*-hvx_128 accumulator_type=int16
+
+$(BIN)/dilate3x3: dilate3x3.cpp $(GENERATOR_DEPS)
+	@-mkdir -p $(BIN)
+	echo "${GENERATOR_DEPS}"
+	$(CXX) $(CXXFLAGS) -O3 -g -fno-rtti $(filter-out %.h,$^) -o $@ $(LDFLAGS)
+
+$(BIN)/%/dilate3x3_cpu.o: $(BIN)/dilate3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f dilate3x3_cpu target=$*
+
+$(BIN)/%/dilate3x3_hvx64.o: $(BIN)/dilate3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f dilate3x3_hvx64 target=$*-hvx_64
+
+$(BIN)/%/dilate3x3_hvx128.o: $(BIN)/dilate3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f dilate3x3_hvx128 target=$*-hvx_128
+
+$(BIN)/median3x3: median3x3.cpp $(GENERATOR_DEPS)
+	@-mkdir -p $(BIN)
+	echo "${GENERATOR_DEPS}"
+	$(CXX) $(CXXFLAGS) -O3 -g -fno-rtti $(filter-out %.h,$^) -o $@ $(LDFLAGS)
+
+$(BIN)/%/median3x3_cpu.o: $(BIN)/median3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f median3x3_cpu target=$*
+
+$(BIN)/%/median3x3_hvx64.o: $(BIN)/median3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f median3x3_hvx64 target=$*-hvx_64
+
+$(BIN)/%/median3x3_hvx128.o: $(BIN)/median3x3
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f median3x3_hvx128 target=$*-hvx_128
+
+$(BIN)/gaussian5x5: gaussian5x5.cpp $(GENERATOR_DEPS)
+	@-mkdir -p $(BIN)
+	echo "${GENERATOR_DEPS}"
+	$(CXX) $(CXXFLAGS) -O3 -g -fno-rtti $(filter-out %.h,$^) -o $@ $(LDFLAGS)
+
+$(BIN)/%/gaussian5x5_cpu.o: $(BIN)/gaussian5x5
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f gaussian5x5_cpu target=$*
+
+$(BIN)/%/gaussian5x5_hvx64.o: $(BIN)/gaussian5x5
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f gaussian5x5_hvx64 target=$*-hvx_64
+
+$(BIN)/%/gaussian5x5_hvx128.o: $(BIN)/gaussian5x5
+	@-mkdir -p $(BIN)/$*
+	$^ -o $(BIN)/$* -e o,h -f gaussian5x5_hvx128 target=$*-hvx_128
+
+$(BIN)/sobel: sobel.cpp $(GENERATOR_DEPS)
```

Comment:
FWIW: you could replace all of the generator-executable-rules with a single template if you followed the same naming convention as test/generators (i.e., foo_generator.cpp -> foo.generator)

      $(BIN)/%.generator: %_generator.cpp $(GENERATOR_DEPS)